tests/cases/fourslash/findAllRefsObjectBindingElementPropertyName06.ts

```diff
@@ -25,5 +25,5 @@ verify.referenceGroups(r2, [
 ]);
 verify.referenceGroups(r4, [
     { definition: "(property) I.property1: number", ranges: [r0, r1, r2, r3] },
-    { definition: "(property) property1: I", ranges: [r4] }
+    { definition: "(property) property1: any", ranges: [r4] }

```

Comment:
`I` was very wrong. `any` is at least the type of `p2`, which is bound to the same symbol as that property because the object literal at this location isn't being correctly interpreted as a binding pattern in all locations. This PR is like a partial fix for that (now the type isn't completely incorrect, at least), but the symbols still need to be bound up correctly. IMO, the root cause is object literal/binding pattern confusion causing us to not handle binding patterns correctly in all the places we need to. I'd like to do a more directed fix at making binding patterns actually appear in expression positions instead of object literals when they're actually binding patterns, and use that to fix this.

For now, this is much better, though.